.github/workflows/sycl_windows_build_and_test.yml

```
@@ -113,20 +102,21 @@ jobs:
       run: |
         cmake --build build --target check-xptifw
     - name: Install
-      if: ${{ always() && !cancelled() && steps.build.outcome == 'success' }}
       shell: bash
       run: |
         cmake --build build --target deploy-sycl-toolchain
+        cmake --build build --target utils/FileCheck/install
+        # Copy FileCheck to install/bin for use by E2E tests.
+        copy %GITHUB_WORKSPACE%\build\bin\FileCheck.exe %GITHUB_WORKSPACE%\install\bin\.
     - name: Upload toolchain
-      if: ${{ always() && !cancelled() && steps.build.outcome == 'success' }}
       uses: actions/upload-artifact@v3
       with:
         name: sycl_windows_default
         path: install
 
   llvm_test_suite:
     needs: build
-    if: ${{ !failure() && inputs.lts_matrix != '[]' }}
+    if: ${{ needs.build.result == 'success' && inputs.lts_matrix != '[]' }}
```

```suggestion
    if: ${{ inputs.lts_matrix != '[]' }}
```
Don't `need: build` (previous line) guarantee that `needs.build.result == 'success'`?